src/vcontrol.rs

```diff
@@ -7,7 +7,8 @@ pub struct VControl<D: Device> {
 }
 
 impl<D: Device> VControl<D> {
-  pub fn new(device: Optolink) -> Self {
+  pub fn new(mut device: Optolink) -> Self {
+    D::Protocol::negotiate(&mut device).expect("protocol negotiation failed");
```

Comment:
Maybe we should rename this to `connect` and return a `Result` instead?

Reply:
Yes, I will have a look on that.

Reply:
Done.